File: scripts/update-mcp-config.js
const fs = require("fs");
const path = require("path");

const file = process.argv[2];

if (!file) {
    console.error("Usage: node update-mcp-config.js <config-json-path>");
    process.exit(1);
}

// Read shared disabled list
const sharedDisabledPath = path.join(__dirname, "..", "shared-disabled-mcp.json");
let sharedDisabled = [];
try {
    const shared = JSON.parse(fs.readFileSync(sharedDisabledPath, "utf8"));
    sharedDisabled = shared.disabledMcpServers || [];
} catch (e) {
    // No shared config, that's fine
}

try {
    let data = {};
    const content = fs.readFileSync(file, "utf8").trim();
    if (content) {
        data = JSON.parse(content);
    }
    if (!data.mcpServers) {
        data.mcpServers = {};
    }
    const defaultServers = {
        playwright: {
            command: "npx",
            args: ["-y", "@playwright/mcp@latest", "--browser", "msedge", "--headless", "--ignore-https-errors", "--isolated", "--output-dir", ".playwright-mcp"]
        },
        context7: {
            type: "sse",
            url: "https://mcp.context7.com/mcp"
        },
        memory: {
            command: "npx",
            args: ["-y", "@modelcontextprotocol/server-memory"]
        },
        "sequential-thinking": {
            command: "npx",
            args: ["-y", "@modelcontextprotocol/server-sequential-thinking"]
        },
        postgres: {
            command: "npx",
            args: ["-y", "@modelcontextprotocol/server-postgres", "postgresql://localhost/postgres"]
        },
        sqlite: {
            command: "npx",
            args: ["-y", "mcp-server-sqlite", "--db", "database.db"]
        },
        docker: {
            command: "npx",
            args: ["-y", "mcp-server-docker"]
        },
        aws: {
            command: "uvx",
            args: ["awslabs.aws-api-mcp-server@latest"]
        }
    };

    const disabledObj = data.disabledMcpServers || {};
    const disabled = new Set([...Object.keys(disabledObj), ...sharedDisabled]);

    let updated = false;

    // Remove servers that are now disabled
    for (const name of disabled) {
        if (data.mcpServers[name]) {
            delete data.mcpServers[name];
            console.log("Removed disabled server: " + name + " from " + file);
            updated = true;
        }
    }

    // Add or update default servers (skip disabled ones)
    for (const [name, config] of Object.entries(defaultServers)) {
        // Skip servers that user has explicitly disabled
        if (disabled.has(name)) {
            continue;
        }
        if (!data.mcpServers[name]) {
            data.mcpServers[name] = config;
            console.log("Initialized " + name + " MCP in " + file);
            updated = true;
        } else {
            // Check playwright configuration details
            if (name === "playwright") {
                const args = data.mcpServers.playwright.args || [];
                let pUpdated = false;
                if (!args.includes("--isolated")) {
                    args.push("--isolated");
                    pUpdated = true;
                }
                const bIdx = args.indexOf("--browser");
                if (bIdx !== -1 && bIdx + 1 < args.length && args[bIdx + 1] !== "msedge") {
                    args[bIdx + 1] = "msedge";
                    pUpdated = true;
                }
                const oIdx = args.indexOf("--output-dir");
                if (oIdx === -1) {
                    args.push("--output-dir", ".playwright-mcp");
                    pUpdated = true;
                } else if (oIdx + 1 < args.length && args[oIdx + 1] !== ".playwright-mcp") {
                    args[oIdx + 1] = ".playwright-mcp";
                    pUpdated = true;
                }
                if (pUpdated) {
                    data.mcpServers.playwright.args = args;
                    console.log("Updated playwright MCP in " + file);
                    updated = true;
                }
            }
        }
    }

    if (updated) {
        fs.writeFileSync(file, JSON.stringify(data, null, 2), "utf8");
    }
} catch (e) {
    console.error("Error updating " + file + ":", e.message);
}

File: scripts/update_mcp_config.py
#!/usr/bin/env python3
import sys
import json
from pathlib import Path

def main():
    if len(sys.argv) < 2:
        print("Usage: python update_mcp_config.py <config-json-path>", file=sys.stderr)
        sys.exit(1)

    config_file = Path(sys.argv[1])

    # Read shared disabled list
    shared_disabled_path = Path(__file__).resolve().parent.parent / "shared-disabled-mcp.json"
    shared_disabled = []
    if shared_disabled_path.exists():
        try:
            shared = json.loads(shared_disabled_path.read_text(encoding="utf-8"))
            shared_disabled = shared.get("disabledMcpServers", [])
        except Exception:
            pass

    try:
        data = {}
        if config_file.exists():
            content = config_file.read_text(encoding="utf-8").strip()
            if content:
                data = json.loads(content)
        
        if "mcpServers" not in data:
            data["mcpServers"] = {}

        default_servers = {
            "playwright": {
                "command": "npx",
                "args": ["-y", "@playwright/mcp@latest", "--browser", "msedge", "--headless", "--ignore-https-errors", "--isolated", "--output-dir", ".playwright-mcp"]
            },
            "context7": {
                "type": "sse",
                "url": "https://mcp.context7.com/mcp"
            },
            "memory": {
                "command": "npx",
                "args": ["-y", "@modelcontextprotocol/server-memory"]
            },
            "sequential-thinking": {
                "command": "npx",
                "args": ["-y", "@modelcontextprotocol/server-sequential-thinking"]
            },
            "postgres": {
                "command": "npx",
                "args": ["-y", "@modelcontextprotocol/server-postgres", "postgresql://localhost/postgres"]
            },
            "sqlite": {
                "command": "npx",
                "args": ["-y", "mcp-server-sqlite", "--db", "database.db"]
            },
            "docker": {
                "command": "npx",
                "args": ["-y", "mcp-server-docker"]
            },
            "aws": {
                "command": "uvx",
                "args": ["awslabs.aws-api-mcp-server@latest"]
            }
        }

        disabled_obj = data.get("disabledMcpServers", {})
        disabled = set(list(disabled_obj.keys()) + shared_disabled)

        updated = False

        # Remove servers that are now disabled
        for name in disabled:
            if name in data["mcpServers"]:
                del data["mcpServers"][name]
                print(f"Removed disabled server: {name} from {config_file}")
                updated = True

        # Add or update default servers (skip disabled ones)
        for name, config in default_servers.items():
            if name in disabled:
                continue

            if name not in data["mcpServers"]:
                data["mcpServers"][name] = config
                print(f"Initialized {name} MCP in {config_file}")
                updated = True
            else:
                if name == "playwright":
                    # Check playwright configuration details
                    args = data["mcpServers"]["playwright"].get("args", [])
                    p_updated = False
                    if "--isolated" not in args:
                        args.append("--isolated")
                        p_updated = True
                    
                    try:
                        b_idx = args.index("--browser")
                        if b_idx + 1 < len(args) and args[b_idx + 1] != "msedge":
                            args[b_idx + 1] = "msedge"
                            p_updated = True
                    except ValueError:
                        pass

                    try:
                        o_idx = args.index("--output-dir")
                        if o_idx + 1 < len(args) and args[o_idx + 1] != ".playwright-mcp":
                            args[o_idx + 1] = ".playwright-mcp"
                            p_updated = True
                    except ValueError:
                        args.extend(["--output-dir", ".playwright-mcp"])
                        p_updated = True

                    if p_updated:
                        data["mcpServers"]["playwright"]["args"] = args
                        print(f"Updated playwright MCP in {config_file}")
                        updated = True

        if updated:
            config_file.write_text(json.dumps(data, indent=2) + "\n", encoding="utf-8")

    except Exception as e:
        print(f"Error updating {config_file}: {e}", file=sys.stderr)

if __name__ == "__main__":
    main()
